### ISSUE TITLE
Make vignette about how to add a new model to the package.

### ISSUE THREAD

**Issue by padpadpadpad** [padpadpadpad/rTPC#65]
Maybe add a vignette showing the walkthrough for how people can add their own model.

Start with a Disclaimer saying "Please maybe don't, there are enough models already and one of them is probably very similar to what you want yours to be".

**Comment by fwimp** [comment]
Agreed. I can probably take a pass at this now I'm back from christmas!

In concert with this is it worthwhile seeing if we can get up some architecture to allow `get_starting_vals()` etc to handle a single file containing a custom model file? It'd allow users to create and test models without going through the furore of actually forking, and would also allow people to define their own models within the rTPC framework easily.

**Comment by padpadpadpad** [comment]
Thinking about this a little more I am still unconvinced rTPC has any huge bespoke architecture that people who want to use their own model formulation need to adhere to.

Apart from `get_start_vals()` `get_lower_lims()` and `get_upper_lims()`, all the functions in `calc_params()` use the model fit and do not need an rTPC function per se, and all the fitting and wrangling is done using **nls.multstart** and the **tidyverse**. I suppose I see the value in showing people how they can define their own function and - if they want - add it to the package, but unlike many R packages there is not much of an rTPC framework. This was partly by design as other people have got much more elegant solutions to wrangling, fitting, and plotting things than me.

**Comment by fwimp** [comment]
Yeah I definitely see that. I mean it shouldn't make a difference to the vast majority of people should we not add it. But also for a select few it could be helpful.

The joy of rTPC is its simplicity in trying to provide just enough to get people running with a bunch of models whilst offloading everything else to other packages.

In the end even just showing what has already been done, and dissecting a model definition in a bit of detail could be helpful.

**Comment by padpadpadpad** [comment]
Made some minor wording changes to the vignette but overall looks great. Will close this now.

**Comment by fwimp** [comment]
Looks really good! Nice edits.

Did notice a mistake though: https://github.com/padpadpadpad/rTPC/blob/5a38f81c13f9aae4fad398b02e787e66994f19d8/vignettes/adding_models.Rmd#L143

Starting values functions are not passed *to* a dataframe, but they are *provided* a dataframe in their arguments which is called `d` internally within the function. The example makes it pretty clear but it's worth just fixing this next time someone has another reason to do a commit (hence I'm just making a note here)

**Comment by padpadpadpad** [comment]
Changed.
